Can't build shared library

Error when linking shared library on Linux x86_64

```
$ cmake -DBUILD_SHARED_LIBS=ON ..
...
$ make
...
[100%] Linking C shared library libsignal-protocol-c.so
/usr/bin/ld: curve25519/CMakeFiles/curve25519.dir/ed25519/fe_isnonzero.c.o: relocation R_X86_64_32 against `.bss' can not be used when making a shared object; recompile with -fPIC
curve25519/CMakeFiles/curve25519.dir/ed25519/fe_isnonzero.c.o: error adding symbols: Bad value
collect2: error: ld returned 1 exit status
src/CMakeFiles/signal-protocol-c.dir/build.make:706: recipe for target 'src/libsignal-protocol-c.so' failed
make[2]: *** [src/libsignal-protocol-c.so] Error 1
CMakeFiles/Makefile2:91: recipe for target 'src/CMakeFiles/signal-protocol-c.dir/all' failed
make[1]: *** [src/CMakeFiles/signal-protocol-c.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/src/curve25519/CMakeLists.txt
+++ b/src/curve25519/CMakeLists.txt
@@ -66,3 +66,8 @@ set(ed25519_SRCS
 )
 
 add_library(curve25519 OBJECT ${curve25519_SRCS} ${ed25519_SRCS})
+
+# Add -fPIC flag
+if(BUILD_SHARED_LIBS)
+	set_property(TARGET curve25519 PROPERTY POSITION_INDEPENDENT_CODE ON)
+endif()

--- a/src/protobuf-c/CMakeLists.txt
+++ b/src/protobuf-c/CMakeLists.txt
@@ -18,3 +18,8 @@ set(protobuf_SRCS
 )
 
 add_library(protobuf-c OBJECT ${protobuf_SRCS})
+
+# Add -fPIC flag
+if(BUILD_SHARED_LIBS)
+	set_property(TARGET protobuf-c PROPERTY POSITION_INDEPENDENT_CODE ON)
+endif()

